app/models/vm_resize_task.rb

```diff
@@ -0,0 +1,45 @@
+class VmResizeTask < MiqRequestTask
+  alias_attribute :vm, :source
+
+  validate :validate_request_type, :validate_state
+
+  AUTOMATE_DRIVES = false
+
+  def self.base_model
+    VmResizeTask
+  end
+
+  def self.get_description(req_obj)
+    name = nil
+    if req_obj.source.nil?
+      # Single source has not been selected yet
+      if req_obj.options[:src_ids].length == 1
+        v = Vm.find_by(:id => req_obj.options[:src_ids].first)
+        name = v.nil? ? "" : v.name
+      else
+        name = "Multiple VMs"
+      end
+    else
+      name = req_obj.source.name
+    end
+
+    new_settings = []
+    new_settings << "Flavor: #{req_obj.options[:flavor_name]}"
+    "#{request_class::TASK_DESCRIPTION} for: #{name} - #{new_settings.join(", ")}"
+  end
+
+  def after_request_task_create
+    update(:description => get_description)
+  end
+
+  def do_request
+    _log.info("Resizing VM #{vm.id}:#{vm.name} to flavor #{options[:flavor_name]}")
+    vm.resize_queue(miq_request.userid, options[:flavor_id])
```

Comment:
Please store the `options[:flavor_id]` as `options[:instance_type]` so it matches how the data is stored from the provisioning dialogs.

Example:
 https://github.com/ManageIQ/manageiq/blob/master/product/dialogs/miq_dialogs/miq_provision_amazon_dialogs_template.yaml#L290

This will assist in quota validation which shares common logic with provisioning.

Reply:
Ahh. I somehow missed this one. Working on it...